pkg/roachprod/vm/aws/aws.go

```diff
@@ -824,13 +827,32 @@ func (p *Provider) runInstance(name string, zone string, opts vm.CreateOpts) err
 	cpuOptions := p.opts.CPUOptions
 
 	// We avoid the need to make a second call to set the tags by jamming
-	// all of our metadata into the TagSpec.
-	tagSpecs := fmt.Sprintf(
-		"ResourceType=instance,Tags=["+
-			"{Key=Lifetime,Value=%s},"+
-			"{Key=Name,Value=%s},"+
-			"{Key=Roachprod,Value=true},"+
-			"]", opts.Lifetime, name)
+	// all of our metadata into the tagSpec.
+	m := vm.GetDefaultLabelMap(opts)
+	var awsLabelMap = map[string]string{
+		vm.TagCluster:   "Cluster",
+		vm.TagCreated:   "Created",
+		vm.TagLifetime:  "Lifetime",
+		vm.TagRoachprod: "Roachprod",
+		"Name":          "Name",
+	}
+	m[vm.TagCreated] = timeutil.Now().Format(time.RFC3339)
+	m["Name"] = name
+
+	var sb strings.Builder
+	sb.WriteString("ResourceType=instance,Tags=[")
+	for key, value := range opts.CustomLabelMap {
+		_, ok := m[strings.ToLower(key)]
+		if ok {
+			return fmt.Errorf("duplicate label name defined: %s", key)
+		}
+		fmt.Fprintf(&sb, "{Key=%s,Value=%s},", key, value)
+	}
+	for key, value := range m {
+		fmt.Fprintf(&sb, "{Key=%s,Value=%s},", awsLabelMap[key], value)
```

Comment:
If `key` is not present in `awsLabelMap`, then `awsLabelMap[key]` will return `""` which is probably not what you want. You need to do something like:

```
if n, ok := awsLabelMap[key]; ok {
    key = n
}
fmt.Fprintf(&sb, "{Key=%s,Value=%s},", key, value)
```